tests/test_config.py

```diff
@@ -1,44 +1,60 @@
 # -*- coding: utf-8 -*-
 
-import json
-from connect.config import Config
 import os
 
+import pytest
+
+from connect.config import Config
+
+
 conf_dict = {
     'apiEndpoint': 'http://localhost:8080/api/public/v1/',
     'apiKey': 'ApiKey XXXX:YYYYY',
     'products': 'CN-631-322-000'
 }
-file_name = 'test_config.json'
 
 
-def _assert_config():
-    assert Config.api_key == conf_dict.get('apiKey')
-    assert Config.api_url == conf_dict.get('apiEndpoint')
-    assert Config.products == [conf_dict.get('products')]
+def setup_module(module):
+    module.prev_dir = os.getcwd()
+    os.chdir(os.path.dirname(os.path.abspath(__file__)))
 
 
-def _reset_config():
+def teardown_module(module):
+    os.chdir(module.prev_dir)
+
+
+def teardown_function():
     Config.api_url, Config.api_key, Config.products = None, None, None
 
 
-def test_init_config_from_json_file():
-    file_path = os.path.join(os.path.dirname(__file__), file_name)
-    with open(file_path, 'w') as output:
-        output.write(json.dumps(conf_dict))
+def test_init_config_with_non_existing_file():
+    with pytest.raises(IOError):
+        Config(file='non_existing_config.json')
 
-    Config(file=file_path)
-    os.remove(file_path)
 
-    _assert_config()
-    _reset_config()
+def test_init_config_with_file():
+    config = Config(file='config.json')
+    _assert_config(config)
 
 
 def test_set_config():
-    Config(
+    config = Config(
         api_key=conf_dict.get('apiKey'),
         api_url=conf_dict.get('apiEndpoint'),
         products=conf_dict.get('products'),
     )
-    _assert_config()
-    _reset_config()
+    _assert_config(config)
+
+
+def _assert_config(config=None):
+    assert Config.api_key == conf_dict.get('apiKey')
+    assert Config.api_url == conf_dict.get('apiEndpoint')
+    assert isinstance(Config.products, list)
+    assert len(Config.products) == 1
+    assert Config.products == [conf_dict.get('products')]
+    if config:
```

Comment:
Config is global
Why send config param in a method `_assert_config`?

Reply:
Is it required for any reason that Config has to be global? I'd like to reduce global state as much as possible. In my tests, I am creating a Config instance and working with that.